src/util/string.h

```diff
@@ -661,14 +661,33 @@ inline const char *bool_to_cstr(bool val)
 	return val ? "true" : "false";
 }
 
+/**
+ * Converts a duration in seconds to a pretty-printed duration in
+ * days, hours, minutes and seconds.
+ *
+ * @param sec duration in seconds (must not be negative)
```

Comment:
if it must not be negative, why does the function handle that case?

Reply:
Negative values seem reasonable, for example for countdowns. If there's already a function it should support also support negative durations.